otter/test/convergence/test_gathering.py

```diff
@@ -213,85 +221,96 @@ class GetScalingGroupServersTests(SynchronousTestCase):
-        fake_response = object()
-        body = {'servers': servers}
-        result = resolve_svcreq(eff, (fake_response, body), *self.req)
-        self.assertEqual(result, {})
-
-    def test_returns_as_servers(self):
-        """
-        Returns servers with AS metadata in it grouped by scaling group ID
-        """
-        as_servers = (
-            [{'metadata': {'rax:auto_scaling_group_id': 'a'}, 'id': i}
-             for i in range(5)] +
-            [{'metadata': {'rax:auto_scaling_group_id': 'b'}, 'id': i}
-             for i in range(5, 8)] +
-            [{'metadata': {'rax:auto_scaling_group_id': 'a'}, 'id': 10}])
-        servers = as_servers + [{'metadata': 'junk'}] * 3
-        eff = resolve_retry_stubs(get_scaling_group_servers())
-        fake_response = object()
-        body = {'servers': servers}
-        result = resolve_svcreq(eff, (fake_response, body), *self.req)
-        self.assertEqual(
-            result,
-            {'a': as_servers[:5] + [as_servers[-1]], 'b': as_servers[5:8]})
-
-    def test_filters_on_user_criteria(self):
-        """
-        Considers user provided filter if provided
-        """
-        as_servers = (
-            [{'metadata': {'rax:auto_scaling_group_id': 'a'}, 'id': i}
-             for i in range(5)] +
-            [{'metadata': {'rax:auto_scaling_group_id': 'b'}, 'id': i}
-             for i in range(5, 8)])
-        servers = as_servers + [{'metadata': 'junk'}] * 3
-        eff = resolve_retry_stubs(
-            get_scaling_group_servers(
-                server_predicate=lambda s: s['id'] % 3 == 0))
-        fake_response = object()
-        body = {'servers': servers}
-        result = resolve_svcreq(eff, (fake_response, body), *self.req)
+        self.now = datetime(2010, 5, 31)
+        self.servers1 = [{'id': 'a', 'a': 'b'}, {'id': 'b', 'b': 'c'}]
+        self.servers2 = [{'id': 'd', 'd': 'e'}]
+        self.freeze = compose(set, map(freeze))
+
+    def _invoke(self):
+        return get_scaling_group_servers(
+            'tid', 'gid', self.now, cache_class=Cache,
+            all_as_servers=intent_func("all-as"),
+            all_servers=intent_func("alls"))
+
+    def _test_no_cache(self, empty):
+        current = [] if empty else self.servers1
+        sequence = SequenceDispatcher([
+            ("cachegstidgid", lambda i: (object(), None)),
+            (("all-as",), lambda i: {} if empty else {"gid": current}),
+            (("cacheistidgid", self.now, current, True), noop)])
+        disp = test_dispatcher(sequence)
+        with sequence.consume():
+            self.assertEqual(
+                sync_perform(disp, self._invoke()), current)
+
+    def test_no_cache(self):
+        """
+        If cache is empty then current list of servers are returned and added
+        to the cache
+        """
+        self._test_no_cache(False)
+        self._test_no_cache(True)
+
+    def _test_old_case(self, last_update, as_srvs=True, cur_srvs=True):
```

Comment:
I don't particularly have an opinion, but I think they may have asked because `test_no_cache` uses the private function `_test_no_cache`, and this is being used by the test `test_old_cache`, but it's named `_test_old_case`